torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -478,35 +588,67 @@ def render(  # type: ignore[override]
         self,
         kernel: CUDATemplateKernel,
         op: "cutlass_gemm_op.GemmOperation",  # type: ignore[name-defined]
-        output_node: Optional[Buffer] = None,
+        template_node: Optional[Buffer] = None,
+        epilogue_nodes: Optional[List[IRNode]] = None,
     ) -> str:
+        if epilogue_nodes is not None and len(epilogue_nodes) > 0:
+            assert (
+                template_node is not None
+            ), "Output node is required for epilogue fusion"
+            assert isinstance(
+                template_node, Buffer
+            ), f"Output node has to be a Buffer, is type {type(template_node)}"
+            assert (
+                template_node.name is not None
+            ), "Output node has to be a Buffer with a name"
+            # This is the name of the output of the Matmul, before epilogues are applied.
+            # it is not necessarily materialized in global memory if we have an epilogue
+            template_output_node_name = template_node.name
+        else:
+            template_output_node_name = None
```

Comment:
Why is it None?

Reply:
From out of memory ( so I hope I don't miss a thing):

The EVT codegen needs to differentiate between auxiliary inputs and the result of the matmul, which is stored in the accumulator, because different operand load expressions are used to retrieve them in the EVT expression. So, if we have epilogue nodes, we need to know the name of the node of the matmul output (template_output_node_name), because if we see a "load(buf1)" instruction, this would translate to "load the accumulator" if it's the matmul output, or into "load auxiliary input buf1", if it's not. 

If we don't have epilogue nodes, then the name of the matmul output is irrelevant since we don't use it. So it's set to None, but that's not really relevant. Maybe I had to initialize it to some value to make the linter happy, I don't quite remember.